src/slangpy_ext/func/tensor.cpp

```
@@ -0,0 +1,870 @@
+            const TensorDesc& desc = tensor.desc();
+            size_t dtype_size = desc.element_layout->stride();
+            size_t byte_offset = desc.offset * dtype_size;
+            size_t buffer_size = tensor.storage()->size() - byte_offset;
+            SGL_CHECK(
+                tensor_bytes <= buffer_size,
+                "Tensor is larger than the tensor storage ({} > {})",
+                tensor_bytes,
+                buffer_size
+            );
+
+            SGL_CU_SCOPE(tensor.storage()->device());
```

Cppcheck (2.20.0)

[style] The function 'emplace_back' is never used.

(unusedFunction)